libs/execution_context/ebpf_native.c

```diff
@@ -263,8 +263,10 @@ ebpf_native_release_reference(_In_opt_ _Post_invalid_ ebpf_native_module_t* modu
 
     ebpf_assert(module->base.marker == _ebpf_native_marker);
 
-    new_ref_count = ebpf_interlocked_decrement_int32(&module->base.reference_count);
-    ebpf_assert(new_ref_count != -1);
+    new_ref_count = ebpf_interlocked_decrement_int64(&module->base.reference_count);
```

Comment:
A similar problem then exists here also. I think we will need to acquire hash table lock here before calling `ebpf_interlocked_decrement_int64()`

Reply:
https://github.com/microsoft/ebpf-for-windows/issues/2326 is tracking the ref-count issue in the native module code. I need to better understand the ref-counting model used by the native module code before I can fix it.

Reply:
For my info -- how did we hit these issues? any manual testing or new tests in CICD?